### PR TITLE
refactor: remove UIKit dependency

### PR DESCRIPTION
**Pull request by jaxtynSong**
## What's improvements?
- Remove UIKit dependency

## how?
- **Create ScrollContext**
ScrollContext provides scroll-related information. 
It is independent of UIKit, but provides a convenient default interface for UIKit

- **Create SushiBeltTrackerItemRect structure**
SushiBeltTrackerItemRect wraps CGRect. 
It is independent of UIKit, but provides a convenient init interface for UIKit

### CODE REVIEW

File: SushiBeltTrackerScrollContext.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+typealias PlatformScrollView = NSScrollView
+
+#endif
+
+public protocol SushiBeltTrackerScrollContext: AnyObject {
+  func scrollDrection() -> SushiBeltTrackerScrollDirection?
+}
+
+extension PlatformScrollView: SushiBeltTrackerScrollContext {
+  public func scrollDrection() -> SushiBeltTrackerScrollDirection? {
```
Comment (GeekTree0101) [inline SushiBeltTrackerScrollContext.swift:29]: I think, `internal` access control is enough!

File: SushiBeltTrackerItemRect.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import AppKit
+
+typealias PlatformView = NSView
+
+#endif
+
+public struct SushiBeltTrackerItemRect {
+  let frameInWindow: CGRect
+}
```
Comment (GeekTree0101) [inline SushiBeltTrackerItemRect.swift:26]: How about make a public initialization such as
```swift
init(x: Float, y: Float, width: Float, height: Float) { ... }
init(x: Int, y: Int, width: Int, height: Int) { ... }
```
is it too much?

File: SushiBeltTrackerScrollContext.swift (hunk 1 above)
Comment (GeekTree0101) [reply SushiBeltTrackerScrollContext.swift:29]: oh. I just see that SushiBeltTrackerScrollContext is public protocol.
well... How about support a default UIScrollView base Context replace from default extension?
for example. 
```swift
public final class SushiBeltTrackerUIScrollContext: SushiBeltTrackerScrollContext {
    weak var scrollView: UIScrollView?
    public init(scrollView: UIScrollView) { ... }
    public func scrollDrection() -> SushiBeltTrackerScrollDirection? { ... }
}
```

File: SushiBeltTrackerItemRect.swift (hunk 2 above)
Comment (ElonPark) [reply SushiBeltTrackerItemRect.swift:26]: I also think it's good to have a `public init`

Comment (jaxtynSong) [reply SushiBeltTrackerItemRect.swift]: good opinion applied it
thx :)

https://github.com/daangn/SushiBelt/pull/9/commits/f6e8e141746110ed107a637f1684a27e817e11e8

File: SushiBeltTrackerScrollContext.swift (hunk 1 above)
Comment (jaxtynSong) [reply SushiBeltTrackerScrollContext.swift]: good opinion applied it
thx :)

https://github.com/daangn/SushiBelt/pull/9/commits/0510e034e7b772dc979225bc7a7b886bdf0bb3e6

**Review by ElonPark — APPROVED** [review]
LGTM 👍 👍

**Review by GeekTree0101 — APPROVED** [review]

**Pull request merged**